doc/nrf/ug_bt_mesh_node_removal.rst

```diff
@@ -0,0 +1,190 @@
+.. _ug_bt_mesh_node_removal:
+
+Removing a node from a mesh network
+###################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+In certain circumstances, there can be a need to remove (unprovision) a node from Bluetooth® mesh network.
+Though the unprovisioning the node is simple, there are some additional steps need to be taken in order to keep security of the network.
+This page describes how to do this properly.
+
+The procedure consists of two steps:
+
+1. Removing the node
+2. Updating a network key on the rest of the nodes.
+
+Removing the node
+*****************
+
+To remove the node from the network and reset its configuration, send the Config Node Reset message from a node running the Configuration Client model, which is usually instantiated on a provisioner.
+
+When using |NCS| API:
+
+1. Call the :c:func:`bt_mesh_cfg_cli_node_reset` function to send Node Reset message to the node to be reset.
+2. Find the node in CDB using :c:func:`bt_mesh_cdb_node_get`.
+3. Remove the node from CDB using :c:func:`bt_mesh_cdb_node_del`.
+
+When using the nRF Mesh mobile app as a provisioner:
+
+1. On the Network screen, tap the mesh device node to be reset.
+2. Scroll down and find :guilabel:`Reset Node` section.
+3. Press on :guilabel:`RESET` button.
+
+Updating a network key
+**********************
+
+Even if the Config Node Reset message has been sent to the node and the Config Node Reset Status message was received, this doesn't guarantee that the node has removed all configuration data.
+To ensure that the removed node is not able to decrypt network messages after being reset, update the network key of the subnet the removed node belonged to.
+
+This procedure can be used in any other cases when a network key needs to be updated.
+For example, when there is a suspicion that the network credentials have been compromised.
+The same procedure can be used to update an application key.
+
+.. note::
+   Updating keys is currently not supported by the nRF Mesh mobile app.
+
+The procedure of updating a network (application) key is described below.
+
+Uploading a new key
+===================
+
+Upload a new network key (or application key) to the nodes by sending the Config NetKey (AppKey) Update message to every node including the node with the Configuration Client model.
+
+The snippet below demonstrates how to do this using |NCS| API:
+
+.. code-block:: c
+
+    static uint8_t send_net_key_update(struct bt_mesh_cdb_node *node, void *user_data)
+    {
+        struct bt_mesh_cdb_subnet *subnet = user_data;
+        uint8_t status = 0;
+        int err;
+
+        err = bt_mesh_cfg_cli_net_key_update(node->net_idx, node->addr, subnet->net_idx,
```

Comment:
```suggestion
        err = bt_mesh_cfg_net_key_update(node->net_idx, node->addr, subnet->net_idx,
```
Guess this applies other places as well.

Reply:
No, without `_cli_` suffix is an old API which will eventually be removed. The new API has `_cli_` suffix. It should be added after upmerge. That is why `DNM` added.

Reply:
Ahh... my bad 👍